assets/src/components/ladder.tsx

```diff
@@ -60,18 +51,69 @@ const Ladder = ({
       ? timepoints.slice().reverse()
       : timepoints
 
+  const timepointStatusY = timepointStatusYFromTimepoints(
+    timepoints,
+    timepointSpacingY
+  )
+  const { ladderVehicles, widthOfLanes } = ladderVehiclesFromVehicles(
+    vehicles,
+    ladderDirection,
+    timepointStatusY
+  )
+
+  const width = 120 + 2 * widthOfLanes * 1.2
+  // (0, 0) is in the center of the first timepoint
+  const viewBox = [-width / 2, -MARGIN_TOP_BOTTOM, width, HEIGHT].join(" ")
+
   return (
-    <svg className="m-ladder" height={HEIGHT} width={WIDTH} viewBox={viewBox}>
-      {vehicles.map((vehicle: Vehicle) => (
-        <LadderVehicle
-          key={vehicle.id}
-          vehicle={vehicle}
-          timepoints={orderedTimepoints}
-          timepointSpacingY={timepointSpacingY}
-          ladderDirection={ladderDirection}
-          selectedVehicleId={selectedVehicleId}
-        />
-      ))}
+    <svg className="m-ladder" height={HEIGHT} width={width} viewBox={viewBox}>
+      {ladderVehicles.map(ladderVehicle => {
```

Comment:
This map function is pretty long, and makes it harder to read the list of everything drawn in the SVG. It could be broken out.

Reply:
Will be fixed in a future PR